src/main/java/com/myhome/services/CommunityAmenityService.java

```diff
@@ -11,4 +12,6 @@
   boolean deleteAmenity(String amenityId);
 
   Set<CommunityAmenity> listAllCommunityAmenities(String communityId);
+
+  boolean updateAmenity(String amenityId, CommunityAmenityDto updatedAmenity);
```

Comment:
Looks like `CommunityAmenityDto` has already `amenityId` property itself. If so, is this `String amenityId` redundant and can be omited?

Reply:
Is the amenity id allowed to be changed? If not I can just setAmenityId to the amenityId from the path variable so that it will always be the ID from the request